tools/data/textrecog/bid_converter.py

```diff
@@ -0,0 +1,251 @@
+
+    return images
+
+
+def load_img_info(files):
+    """Load the information of one image.
+    Args:
+        files (tuple): The tuple of (img_file, groundtruth_file)
+    Returns:
+        img_info (dict): The dict of the img and annotation information
+    """
+    assert isinstance(files, tuple)
+
+    img_file, gt_file = files
+    assert osp.basename(gt_file).split('_')[0] == osp.basename(gt_file).split(
+        '_')[0]
+    # read imgs while ignoring orientations
+    img = mmcv.imread(img_file, 'unchanged')
+
+    img_info = dict(
+        file_name=osp.basename(img_file),
+        height=img.shape[0],
+        width=img.shape[1],
+        segm_file=osp.basename(gt_file))
+
+    if osp.splitext(gt_file)[1] == '.txt':
+        img_info = load_txt_info(gt_file, img_info)
+    else:
+        raise NotImplementedError
+
+    return img_info
+
+
+def load_txt_info(gt_file, img_info):
+    """Collect the annotation information.
+
+    The annotation format is as the following:
+    x, y, w, h, text
+    977, 152, 16, 49, NOME
+    962, 143, 12, 323, APPINHANESI BLAZEK PASSOTTO
+    906, 446, 12, 94, 206940361
+    905, 641, 12, 44, SPTC
+
+    Args:
+        gt_file (str): The path to ground-truth
+        img_info (dict): The dict of the img and annotation information
+
+    Returns:
+        img_info (dict): The dict of the img and annotation information
+    """
+
+    with open(gt_file, 'r', encoding='latin1') as f:
+        anno_info = []
+        for line in f:
+            line = line.strip('\n')
+            # Ignore hard samples
+            if line[0] == '[' or line[0] == 'x':
+                continue
+            ann = line.split(',')
+            bbox = ann[0:4]
+            bbox = [int(_) for _ in bbox]
+            x, y, w, h = bbox
+            # in case ',' exists in label
+            word = ','.join(ann[4:]) if len(ann[4:]) > 1 else ann[4]
+            # remove the initial space
+            word = word.strip()
+            bbox = [x, y, x + w, y, x + w, y + h, x, y + h]
+
+            anno = dict(bbox=bbox, word=word)
+            anno_info.append(anno)
+
+    img_info.update(anno_info=anno_info)
+
+    return img_info
+
+
+def split_train_val_list(full_list, val_ratio):
+    """Split list by val_ratio
+
+    Args:
+        full_list (list): List to be splited
+        val_ratio (float): Split ratio for val set
+
+    return:
+        list(list, list): Train_list and test_list
+    """
+
+    n_total = len(full_list)
+    offset = int(n_total * val_ratio)
+    if n_total == 0 or offset < 1:
+        return [], full_list
+    val_list = full_list[:offset]
+    train_list = full_list[offset:]
+    return [train_list, val_list]
+
+
+def generate_ann(root_path, image_infos, preserve_vertical, val_ratio, format):
+    """Generate cropped annotations and label txt file.
+    Args:
+        root_path (str): The root path of the dataset
+        split (str): The split of dataset. Namely: training or test
```

Comment:
incorrect docstring.

Reply:
> incorrect docstring.
split (str) should be deleted from the docstring since there is no such param input to this function